Update qscout tutorial notebook

bharat-thotakura opened the pull request:
Address #463 as a sample notebook. Mainly added more descriptions to the existing notebooks, sometimes pulling from existing examples. The css and qss versions of the notebook are modeled to be nearly identical.

bharat-thotakura commented on docs/source/optimizations/qscout/qscout_css.ipynb:
```diff
@@ -1,6 +1,7 @@
 {
```
I thought I'd change it up by naming it qbits (like cbits) haha, but qubits is better for consistency. Fixed in 95caeed

stephanielee9 commented:
@bharat-thotakura a couple nits:
1. There are a few "lets" that I think should be "let's"
2. I think it would be nice for the `qscout.rst` file to have a brief description before listing the links
3. Remove "For those who’d like to follow along, you can try the code out on our [Jupyter Hub](https://tinyurl.com/2p9dtyxj)."

Why does the second code block have the user balance?

bharat-thotakura commented:
> @bharat-thotakura a couple nits:
> 
> 1. There are a few "lets" that I think should be "let's"
> 2. I think it would be nice for the `qscout.rst` file to have a brief description before listing the links
> 3. Remove "For those who’d like to follow along, you can try the code out on our [Jupyter Hub](https://tinyurl.com/2p9dtyxj)."
> 
> Why does the second code block have the user balance?

1.  Typos fixed
2.  Sure, do you mean the qscout links or installing qiskit/cirq-superstaq? And do you have any suggestions on the text that should definitely be there or would be particularly helpful?
3. That was a remnant of the old existing notebooks, now removed.
4. I saw that the AQT docs notebooks had it, so I thought to maybe include it. But if the notebook is primarily for viewing on docs then it's probably not necessary. Removed now.

bharat-thotakura commented:
@stephanielee9 

I have updated `docs/source/optimizations/qscout/qscout.rst` with a template introductory description, but I'm not sure I have enough knowledge on QSCOUT to provide a concise description of our exact engagement/relationship with QSCOUT. Maybe @richrines1 can help with that? Currently, I've left a placeholder [INSERT] for it.

As for the "Running cells with '/bin/python3.10' requires the ipykernel package" message, I think I've removed that error description in the notebook content. Seems to have been a byproduct of changing from my local kernel to a standard Python kernel that does not have ipykernel installed.

stephanielee9 commented:
Nice! If @richrines1 has anything to add to the description, that'd be great, but otherwise it looks good for now. Could you update a couple things?
* It may just be my IDE but I think there's an extra line break between "simulate" and "circuits" in "how to get started with using Superstaq to compile, optimize, and simulate circuits". I think also one right before "By the end, we will..."
* For "We will go over the process of setting up a superstaq provider", maybe capitalize Superstaq or make it inline code?
* The links to the notebooks seem to be in a weird spot right now / a little bit hard to notice. Maybe move the `... toctree::`` section between the last sentence of of the description and the References heading and / or give the toctree a heading
* Move the References to the `h2` heading level (it's currently showing up in the sidebar) (e.g., use `-` instead of `=` under the title)

bharat-thotakura commented:
@stephanielee9 

Fixed the layout issues you brought up in 4b6949a. Let me know if there's anything else.

stephanielee9 reviewed (approved):
lgtm!!

richrines1 commented on docs/source/optimizations/qscout/qscout_css.ipynb:
```diff
@@ -30,11 +32,14 @@
-    "## Imports and Token"
+    "## Imports and API Token\n",
+    "\n",
+    "For this tutorial, we will just need the imports listed below to run the Superstaq optimizations. If you do not have `cirq-superstaq` installed on your machine, you can follow the guide [here](https://docs-superstaq.readthedocs.io/en/latest/get_started/installation.html#installing-cirq-superstaq) to set it up and its dependencies. "
```
imo we should lead with something like "this notebook uses cirq-superstaq, our superstaq client for Cirq; you can try it out by running pip install cirq-superstaq" instead of making it just one of the "imports listed below"

bharat-thotakura replied:
I initially assumed that users would go to the installation guide first, but I've added your suggestion to be more self-contained.

richrines1 commented on docs/source/optimizations/qscout/qscout_qss.ipynb:
```diff
@@ -30,111 +32,143 @@
-    "## Imports and Token"
+    "## Imports and API Token\n",
+    "\n",
+    "For this tutorial, we will just need the imports listed below to run the Superstaq optimizations. If you do not have `qiskit-superstaq` installed on your machine, you can follow the guide [here](https://docs-superstaq.readthedocs.io/en/latest/get_started/installation.html#installing-qiskit-superstaq) to set it up and its dependencies. "
```
i think we also talked about adding something like "this example notebook uses qiskit-superstaq; you can try it out by running pip install qiskit-superstaq[examples]" to the top of qiskit-superstaq notebooks

bharat-thotakura commented:
@stephanielee9 @richrines1 

I've added a pip install cell to the notebooks, if that looks fine then this PR can probably be merged now.

richrines1 commented:
sorry two more things:
1. qscout assumes we measure every qubit in the circuit so we should probably do so in all of the example circuits (currently a few of the example circuits have measurements on only one of two qubits, but this conflicts with the jaqal programs (which end with "measure_all"))
2. in the examples in which swap mirroring actually inserts a swap, can we add a line like "note that the classical bits (qiskit) / keys (cirq) associated with each measurement have been swapped in the compiled circuit"?

richrines1 reviewed (approved):
awesome, thanks :) lgtm!
